src/test/java/de/fraunhofer/aisec/cpg/enhancements/variableResolution/VariableResolverCppTest.java

```diff
@@ -0,0 +1,433 @@
+
+    List<RecordDeclaration> classes = Util.filterCast(nodes, RecordDeclaration.class);
+
+    // Inner class and its fields
+    innerClass =
+        Util.getOfTypeWithName(nodes, RecordDeclaration.class, "ScopeVariables::InnerClass");
+    innerVarName = Util.getSubnodeOfTypeWithName(innerClass, FieldDeclaration.class, "varName");
+    innerStaticVarName =
+        Util.getSubnodeOfTypeWithName(innerClass, FieldDeclaration.class, "staticVarName");
+    innerImpThis = Util.getSubnodeOfTypeWithName(innerClass, FieldDeclaration.class, "this");
+
+    main = Util.getOfTypeWithName(nodes, FunctionDeclaration.class, "main");
+
+    outer_function1 =
+        outerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function1"))
+            .collect(Collectors.toList())
+            .get(0);
+    forStatements = Util.filterCast(SubgraphWalker.flattenAST(outer_function1), ForStatement.class);
+
+    // Functions i nthe outer and inner object
+    outer_function2 =
+        outerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function2"))
+            .collect(Collectors.toList())
+            .get(0);
+    outer_function3 =
+        outerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function3"))
+            .collect(Collectors.toList())
+            .get(0);
+    outer_function4 =
+        outerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function4"))
+            .collect(Collectors.toList())
+            .get(0);
+    outer_function5 =
+        outerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function5"))
+            .collect(Collectors.toList())
+            .get(0);
+    inner_function1 =
+        innerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function1"))
+            .collect(Collectors.toList())
+            .get(0);
+    inner_function2 =
+        innerClass.getMethods().stream()
+            .filter(method -> method.getName().equals("function2"))
+            .collect(Collectors.toList())
+            .get(0);
+
+    for (CallExpression call : calls) {
+      Expression first = call.getArguments().get(0);
+      String logId = ((Literal) first).getValue().toString();
+
+      Expression second = call.getArguments().get(1);
+      callParamMap.put(logId, second);
+    }
+  }
+
+  public DeclaredReferenceExpression getCallWithReference(String literal) {
+    Expression exp = callParamMap.get(literal);
+    if (exp instanceof DeclaredReferenceExpression) return (DeclaredReferenceExpression) exp;
+    return null;
+  }
+
+  public MemberExpression getCallWithMemberExpression(String literal) {
+    Expression exp = callParamMap.get(literal);
+    if (exp instanceof MemberExpression) return (MemberExpression) exp;
+    return null;
+  }
+
+  @Test
```

Comment:
I am not sure if I like the test-design of this test class. Having just one assert in a test is unnecessary overhead of launching a test in my opinion

Reply:
It makes it more modular to se which cases exactly fail for the test-first approach, after all tests run properly we can consider packing them into one class.